spec/rails_helper.rb

```diff
@@ -213,16 +213,25 @@ def seed_base_data_for_tests
   end
 
   config.before(:all) do
-    unless Thread.current[:skipped_last_seeding]
+    seed_current = self.class.metadata[:skip_seed].nil? || self.class.metadata[:skip_seed] == false
+    seeded_last = Thread.current[:seeded_last]
+
+    if seeded_last && !seed_current
       DatabaseCleaner.clean_with(:truncation)
     end
 
-    Thread.current[:skipped_last_seeding] = true
-    unless self.class.metadata[:skip_seed]
+    if seeded_last && seed_current
+      define_global_variables
```

Comment:
You can reorganize this a bit like so:

```
if seeded_last && !seed_current
  DatabaseCleaner.clean_with(:truncation) # remove previously seeded data
end
if seed_current
  seed_base_data_for_tests if !seeded_last # need to re-seed data
  define_global_variables
end

Reply:
I was doing it the other way do be explicit about all the cases, also why I added the comment about the !seeded && !seeded case. You think this is better?